### ISSUE TITLE
feat: Add all scikit-learn scores to report metrics string accepted

### ISSUE THREAD

**Issue by MarieSacksick** [probabl-ai/skore#1519]
### Is your feature request related to a problem? Please describe.

I wanted to do the following code, but couldn't:  

```
datasets = fetch_employee_salaries()
X, y = datasets.X, datasets.y
X_train, X_test, y_train, y_test = train_test_split(X, y)
baseline_dummy = DummyRegressor()
baseline_dumb_est = EstimatorReport(
    baseline_dummy, X_train=X_train, y_train=y_train, X_test=X_test, y_test=y_test
)
baseline_dumb_cv.metrics.report_metrics(scoring=["neg_max_error"])
```

But I got the following (and clear) error:   
![Image](https://github.com/user-attachments/assets/a0dc8ce3-9869-452f-bea0-928dd0fb1c2c)

### Describe the solution you'd like

I'd like to be able to give any pre-defined metric in scikit-learn:    
https://scikit-learn.org/1.5/modules/model_evaluation.html

**Comment by auguste-probabl** [comment]
Also, all `neg_*` metrics should have their equivalent without `neg_`.

**Comment by Muhammad-Rebaal** [comment]
Hi @MarieSacksick !

Could you assign this issue to me ? I'd like to solve it .

Thank You !

**Comment by glemaitre** [comment]
I'm unsure if we should accept the `neg_*` one. This convention does not make sense here. Potentially, we could detect the `neg_*` and raise an error stating that you have to pass the positive counter part.

Also, we need to be careful by supporting the feature: we need to make sure that the caching mechanism is working as expected. We should therefore check if we should reimplement the different functions to make sure to call the cached predictions or if we should come with some trick where we get the scorer and tweak it.
